drivers/spi/spi_stm32.c

```diff
@@ -459,129 +500,85 @@ static int spi_stm32_init(struct device *dev)
+ *                                 'SPI', x, 'BASE_ADDRESS')
+ *     _irq = codegen.device_tree_controller_property('SPI', x, 'IRQ_0')
+ *     _irq_prio = codegen.device_tree_controller_property(
+ *                                 'SPI', x, 'IRQ_0_PRIORITY')
+ *     _clock_enr = codegen.device_tree_controller_property(
+ *                                 'SPI', x, 'clocks_BITS')
+ *     _clock_bus = codegen.device_tree_controller_property(
+ *                                 'SPI', x, 'clocks_BUS')
+ *     _config_irq = codegen.device_tree_controller_data_name(
+ *                                 'SPI', x, 'config_irq')
+ *     _config_info = codegen.device_tree_controller_data_name(
+ *                                 'SPI', x, 'config')
+ *     _data = codegen.device_tree_controller_data_name('SPI', x, 'data')
+ *     _dev_name = codegen.device_tree_controller_device_name('SPI', x)
+ *     _drv_name = codegen.device_tree_controller_driver_name('SPI', x)
+ *     #
+ *     # config irq
+ *     if codegen.config_property('CONFIG_SPI_STM32_INTERRUPT'):
+ *         codegen.outl('DEVICE_DECLARE({});'.format(_dev_name))
+ *         codegen.outl('static void {}(void)'.format(_config_irq))
+ *         codegen.outl('{')
+ *         codegen.outl('\tIRQ_CONNECT( \\')
+ *         codegen.outl('\t\t{}, \\'.format(_irq))
+ *         codegen.outl('\t\t{}, \\'.format(_irq_prio))
+ *         codegen.outl('\t\t{}, \\'.format(_isr))
+ *         codegen.outl('\t\tDEVICE_GET({}), \\'.format(_dev_name))
+ *         codegen.outl('\t\t0);')
+ *         codegen.outl('\tirq_enable({});'.format(_irq))
+ *         codegen.outl('}')
+ *     #
+ *     # config info
+ *     codegen.outl('static const struct spi_stm32_config {} = {{'.format(
+ *         _config_info))
+ *     codegen.outl('\t.spi = (SPI_TypeDef *){},'.format(_base_address))
+ *     codegen.outl('\t.pclken = {')
+ *     codegen.outl('\t\t.enr = {},'.format(_clock_enr))
+ *     codegen.outl('\t\t.bus = {},'.format(_clock_bus))
+ *     codegen.outl('\t},')
+ *     if codegen.config_property('CONFIG_SPI_STM32_INTERRUPT'):
+ *         codegen.outl('\t.config_irq = {},'.format(_config_irq))
+ *     codegen.outl('};')
+ *     #
+ *     # data
+ *     codegen.outl('static struct spi_stm32_data {} = {{'.format(_data))
+ *     codegen.outl('\tSPI_CONTEXT_INIT_LOCK({}, ctx),'.format(_data))
+ *     codegen.outl('\tSPI_CONTEXT_INIT_SYNC({}, ctx),'.format(_data))
+ *     codegen.outl('};')
+ *     #
+ *     # device init
+ *     codegen.outl('DEVICE_AND_API_INIT( \\')
+ *     codegen.outl('\t{}, \\'.format(_dev_name))
+ *     codegen.outl('\t{}, \\'.format(_drv_name))
+ *     codegen.outl('\t{}, \\'.format(_device_init))
+ *     codegen.outl('\t&{}, \\'.format(_data))
+ *     codegen.outl('\t&{}, \\'.format(_config_info))
+ *     codegen.outl('\t{}, \\'.format(_level))
+ *     codegen.outl('\t{}, \\'.format(_prio))
+ *     codegen.outl('\t&{});'.format(_api))
```

Comment:
same for this device init block

Reply:
Added that to my todo list.

Reply:
> Shall the expanded code use the device macros from zephyr or expand to a pure C representation?

According to the new usage rulesthe  preprocessor does have precedence before codegen. So the macros shall be sued.